src/functional.ts

```diff
@@ -228,16 +172,11 @@ export function functionalInterfaceFactory({
    * https://ethresear.ch/t/fast-verification-of-multiple-bls-signatures/5407
    */
   async function asyncVerifyMultipleSignatures(sets: SignatureSet[]): Promise<boolean> {
-    if (implementation === "herumi") return verifyMultipleSignatures(sets);
     try {
-      return blst.asyncVerifyMultipleAggregateSignatures(
-        sets.map((set) => ({
-          message: set.message,
-          publicKey: convertToBlstPublicKeyArg(set.publicKey),
-          signature: convertToBlstSignatureArg(set.signature),
-        }))
-      );
-    } catch {
+      if (implementation === "herumi") return Signature.verifyMultipleSignatures(sets);
+      return Signature.asyncVerifyMultipleSignatures(sets);
```

Comment:
`try/catch` does not work well if this is async
```suggestion
      return await Signature.asyncVerifyMultipleSignatures(sets);
```

Reply:
Very good catch sir! Updated all of the async functions with `await`